homeassistant/components/tesla/__init__.py

```diff
@@ -145,12 +148,23 @@ async def async_setup_entry(hass, config_entry):
         _LOGGER.error("Unable to communicate with Tesla API: %s", ex.message)
         return False
     _async_save_tokens(hass, config_entry, access_token, refresh_token)
+    coordinator = TeslaDataUpdateCoordinator(
+        hass, config_entry=config_entry, controller=controller
+    )
+    # Fetch initial data so we have data when entities subscribe
     entry_data = hass.data[DOMAIN][config_entry.entry_id] = {
         "controller": controller,
+        "coordinator": coordinator,
         "devices": defaultdict(list),
         DATA_LISTENER: [config_entry.add_update_listener(update_listener)],
     }
     _LOGGER.debug("Connected to the Tesla API")
+
+    await coordinator.async_refresh()
+    
```

Comment:
```suggestion
```

Reply:
Yah fixed it already. Thanks.

Reply:
sorry github is bad about adding the extra whitespace